File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(), happiness.end(), greater<int>());
        
        long long total = 0;
        
        for (int i = 0; i < k; i++) {
            int contribution = max(0, happiness[i] - i);
            total += contribution;
            
            if (contribution == 0) {
                break;
            }
        }
        
        return total;
    }
};

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.py
class Solution(object):
    def maximumHappinessSum(self, happiness, k):

        happiness.sort(reverse=True)
        
        total = 0
        
        for i in range(k):
            contribution = max(0, happiness[i] - i)
            total += contribution
            
            if contribution == 0:
                break
        
        return total